src/main/java/org/sasanlabs/service/vulnerability/xss/reflected/XSSInImgTagAttribute.java

```diff
@@ -168,18 +176,18 @@ public UrlParamBasedImgTagAttrInjection() {
             variant = Variant.SECURE,
             htmlTemplate = "LEVEL_1/XSS")
     public ResponseEntity<String> getVulnerablePayloadLevelSecure(
-            @RequestParam Map<String, String> queryParams) {
+            @RequestParam(PARAMETER_NAME) String imageLocation) {
         String vulnerablePayloadWithPlaceHolder = "<img src=\"%s\" width=\"400\" height=\"300\"/>";
-        StringBuilder payload = new StringBuilder();
 
-        for (Map.Entry<String, String> map : queryParams.entrySet()) {
-            if (allowedValues.contains(map.getValue())) {
-                payload.append(
-                        String.format(
-                                vulnerablePayloadWithPlaceHolder,
-                                StringEscapeUtils.escapeHtml4(map.getValue())));
-            }
+        if (imageLocation.startsWith(IMAGE_RESOURCE_PATH)
+                && imageLocation.endsWith(FILE_EXTENSION)) {
+            String payload =
+                    String.format(
+                            vulnerablePayloadWithPlaceHolder,
+                            HtmlUtils.htmlEscapeHex(imageLocation));
```

Comment:
Here even StringEscapeUtils.escapeHtml4 could work right?

Reply:
Basically that should work, too, but the OWASP suggests to use HTML Attribute Encoding: https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#output-encoding-rules-summary